### ISSUE TITLE
Why call self.attention.forward

### ISSUE THREAD

**Issue by mst272** [meta-llama/llama#1055]
I am learning llama, where self.attention.forward is explicitly called. But normally, we don’t write code that explicitly calls forward, but why would llama authors explicitly call it here? Is it just because of habit?

Explicitly called here: h = x + self.attention.forward(self.attention_norm(x), start_pos, freqs_cis, mask)

```
class TransformerBlock(nn.Module):
    def __init__(self, layer_id: int, args: ModelArgs):
        super().__init__()
        self.n_heads = args.n_heads
        self.dim = args.dim
        self.head_dim = args.dim // args.n_heads
        self.attention = Attention(args)
        self.feed_forward = FeedForward(
            dim=args.dim,
            hidden_dim=4 * args.dim,
            multiple_of=args.multiple_of,
            ffn_dim_multiplier=args.ffn_dim_multiplier,
        )
        self.layer_id = layer_id
        self.attention_norm = RMSNorm(args.dim, eps=args.norm_eps)
        self.ffn_norm = RMSNorm(args.dim, eps=args.norm_eps)

    def forward(
        self,
        x: torch.Tensor,
        start_pos: int,
        freqs_cis: torch.Tensor,
        mask: Optional[torch.Tensor],
    ):
        h = x + self.attention.forward(
            self.attention_norm(x), start_pos, freqs_cis, mask
        )
        out = h + self.feed_forward.forward(self.ffn_norm(h))
        return out
```

**Comment by subramen** [comment]
That's a good call-out, the best practice is to use the module's `__call__`

In this case, the change is no-op since we aren't using hooks in the model. In case you are interested in updating the code to reflect this best-practice I'd be happy to review a PR!

**Comment by mst272** [comment]
> That's a good call-out, the best practice is to use the module's `__call__`
> 
> In this case, the change is no-op since we aren't using hooks in the model. In case you are interested in updating the code to reflect this best-practice I'd be happy to review a PR!

Thanks for your response.
 I've submitted a PR addressing the issue based on your suggestion. Looking forward to your review!

**Comment by subramen** [comment]
Thanks for helping to improve our code quality @mst272, PR is merged
